src/main/java/gov/usgs/wma/waterdata/LoadDiscreteGroundWater.java

```diff
@@ -0,0 +1,90 @@
+package gov.usgs.wma.waterdata;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+public class LoadDiscreteGroundWater implements Function<RequestObject, ResultObject> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(LoadDiscreteGroundWater.class);
+
+	public static final String STATUS_SUCCESS = "success";
+	public static final String STATUS_FAIL = "fail";
+	public static final String STATUS_SUCCESS_MESSAGE = "Successfully inserted gw levels with field visit identifiers: %s";
+	public static final String MESSAGE_NO_RECORDS = "No records found for field visit identifiers: %s";
+	public static final String FAIL_MESSAGE_NULL_IDENTIFIER = "field visit identifier was null";
+	public static final String FAIL_MESSAGE_INSERT_FAILED = "Selected row count: %s and inserted row count: %s differ, insert failed for field visit identifiers: %s";
+
+	private final TransformDao transformDao;
+	private final ObservationDao observationDao;
+
+	@Autowired
+	public LoadDiscreteGroundWater(TransformDao transformDao, ObservationDao observationDao) {
+		this.transformDao = transformDao;
+		this.observationDao = observationDao;
+	}
+
+	@Override
+	public  ResultObject apply(RequestObject request) {
+		ResultObject result = processRequest(request);
+		if (STATUS_FAIL.equalsIgnoreCase(result.getStatus())) {
+			throw new RuntimeException(result.getFailMessage());
+		} else {
+			return result;
+		}
+	}
+
+	protected ResultObject processRequest(RequestObject request) {
+
+		List<String> fieldVisitIdentifiers = request.getFieldVisitIdentifiers();
```

Comment:
I find this very confusing. It appears that we get a list of field visits and each field visit has a list of ground water values? If that is correct, it might be clearer to only process one field visit at a time. If it is one-to-one (field visit to ground water), is there a way to make that more obvious?

Reply:
The transformer passes on a list of `fieldVisitIdentifiers` to the loader, yes.